### PR TITLE
feat(core): add support for autocomplete on bool value

### LINKED ISSUE #919: Add autocomplete on boolean value
This probably requires getting the type from the arg spec.

This means we should have a standard function that extract via reflection the type of a variable.

### PR DESCRIPTION
**Pull request by jerome-quere**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #919

### CODE REVIEW

File: internal/core/autocomplete_test.go [hunk 1]
```diff
@@ -135,8 +140,10 @@ func TestAutocomplete(t *testing.T) {
 	t.Run("scw test flower delete with-leaves=true ", run(&testCase{Suggestions: AutocompleteSuggestions{"anemone", "hibiscus"}})) // invalid notation
 	t.Run("scw test flower delete hibiscus n", run(&testCase{Suggestions: nil}))
 	t.Run("scw test flower delete hibiscus w", run(&testCase{Suggestions: AutocompleteSuggestions{"with-leaves="}}))
-	t.Run("scw test flower delete hibiscus with-leaves=true", run(&testCase{Suggestions: nil}))
+	t.Run("scw test flower delete hibiscus with-leaves=true", run(&testCase{Suggestions: AutocompleteSuggestions{"with-leaves=true"}}))
```
Comment (kindermoumoute) [inline internal/core/autocomplete_test.go:143]: this should still be nil to be iso with previous test on line 136, 129, 120,...

Comment (jerome-quere) [reply internal/core/autocomplete_test.go:143]: Other tests you list are using `WordToCompleteIndex` this is not the case for this one

Comment (kindermoumoute) [reply internal/core/autocomplete_test.go:143]: True! We lack of a test similar to this one for enums:
```
t.Run("scw test flower create leaves.0.size=L", run(&testCase{Suggestions: AutocompleteSuggestions{"leaves.0.size=L"}}))
```

File: internal/core/autocomplete_test.go [hunk 2]
```diff
@@ -135,8 +141,10 @@ func TestAutocomplete(t *testing.T) {
...
 	t.Run("scw test flower delete hibiscus with-leaves=true ", run(&testCase{Suggestions: AutocompleteSuggestions{"anemone"}}))
+	t.Run("scw test flower delete hibiscus with-leaves=", run(&testCase{Suggestions: AutocompleteSuggestions{"with-leaves=false", "with-leaves=true"}}))
+	t.Run("scw test flower delete hibiscus with-leaves=tr", run(&testCase{Suggestions: AutocompleteSuggestions{"with-leaves=true"}}))
```
Comment (kindermoumoute) [inline internal/core/autocomplete_test.go:147]: add a test with invalid completion
```
t.Run("scw test flower delete hibiscus with-leaves=yes", run(&testCase{Suggestions: nil}))
```

**Pull request merged**